pkgs/tools/text/vale/styles.nix

```diff
@@ -0,0 +1,118 @@
+{ lib, stdenvNoCC, fetchFromGitHub }:
+
+let
+  buildStyle =
+    { name
+    , stylePath ? name
+    , ...
+    }@args:
+    stdenvNoCC.mkDerivation ({
+      pname = "vale-style-${name}";
+      installPhase = ''
+        runHook preInstall
+        install -Dm644 ${stylePath}/*.yml -t "$out/share/vale/styles/${name}"
+        runHook postInstall
+      '';
+      meta = {
+        platforms = lib.platforms.all;
+      } // (args.meta or { });
+    } // removeAttrs args [ "meta" "name" ]);
+in
+{
+  alex = buildStyle rec {
+    name = "alex";
+    version = "0.2.1";
+    src = fetchFromGitHub {
+      owner = "errata-ai";
+      repo = "alex";
+      rev = "v${version}";
+      hash = "sha256-xNF7se2FwKgNe5KYx/zvGWpIwBsBADYGH4JV1lUww+Q=";
+    };
+    meta.license = lib.licenses.mit;
+  };
+
+  google = buildStyle rec {
+    name = "Google";
+    version = "0.6.0";
+    src = fetchFromGitHub {
+      owner = "errata-ai";
+      repo = "Google";
+      rev = "v${version}";
+      hash = "sha256-TQS/hgS6tEWPSuZpEbX65MdYSE/+HJVcnzIuQbhIG2M=";
+    };
+    meta.license = lib.licenses.mit;
+  };
+
+  joblint = buildStyle rec {
+    name = "Joblint";
+    version = "0.4.1";
+    src = fetchFromGitHub {
+      owner = "errata-ai";
+      repo = "Joblint";
+      rev = "v${version}";
+      hash = "sha256-zRz5ThOg5RLTZj3dYPe0PDvOF5DjO31lduSpi2Us87U=";
+    };
+    meta.license = lib.licenses.mit;
+  };
+
+  microsoft = buildStyle rec {
+    name = "Microsoft";
+    version = "0.14.0";
+    src = fetchFromGitHub {
+      owner = "errata-ai";
+      repo = "Microsoft";
+      rev = "v${version}";
+      hash = "sha256-22rGNLZOsWYQ+H3CcM2b1zOXV3kNPcgYqDpaCg1iv9o=";
+    };
+    meta.license = lib.licenses.mit;
+  };
+
+  proselint = buildStyle rec {
+    name = "proselint";
+    version = "0.3.3";
+    src = fetchFromGitHub {
+      owner = "errata-ai";
+      repo = "proselint";
+      rev = "v${version}";
+      hash = "sha256-faeWr1bRhnKsycJY89WqnRv8qIowUmz3EQvDyjtl63w=";
+    };
+    meta.license = lib.licenses.bsd3;
+  };
+
+  readability = buildStyle rec {
+    name = "readability";
```

Comment:
Result of `nixpkgs-review pr 301204` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>valeStyles.readability</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>valeStyles.alex</li>
    <li>valeStyles.google</li>
    <li>valeStyles.joblint</li>
    <li>valeStyles.microsoft</li>
    <li>valeStyles.proselint</li>
    <li>valeStyles.redhat</li>
    <li>valeStyles.write-good</li>
  </ul>
</details>

```suggestion
    name = "Readability";
```